primitives/src/unified_num.rs

```diff
@@ -961,6 +955,69 @@ mod test {
             );
         }
     }
+
+    #[test]
+    fn test_unified_num_mod_floor_gcd_lcm_divides_is_multiple_of_div_rem() {
```

Comment:
I'm uncertain about these tests. You're actually checking `0.00000008` value in the first test for example, not integers so the math is different for floating numbers.
Another things I realised is that `num::Integer` allows these methods and it's not implemented for floating numbers, so maybe we don't need to include `Integer` impl for `UnitfiedNum`